crates/subspace-networking/src/request_handlers/generic_request_handler.rs

```diff
@@ -29,42 +27,38 @@ use tracing::{debug, trace};
 // Could be changed after the production feedback.
 const REQUESTS_BUFFER_SIZE: usize = 50;
 
-/// Defines a config for the generic request handler for the request-response protocol.
-pub struct RequestHandlerConfig<Req, Resp> {
-    /// Tracing log target
-    pub log_target: &'static str,
-    /// Request-response protocol name
-    pub protocol_name: &'static str,
-    /// Actual request-response handler.
-    pub request_handler: ExternalRequestHandler<Req, Resp>,
+/// Generic request with associated response
+pub trait GenericRequest: Encode + Decode + 'static {
+    const PROTOCOL_NAME: &'static str;
+    const LOG_TARGET: &'static str;
+    /// Response type that corresponds to this request
+    type Response: Encode + Decode + 'static;
 }
 
-/// Type alias for the actual external request handler.
-pub type ExternalRequestHandler<Req, Resp> =
-    Arc<dyn (Fn(&Req) -> Option<Resp>) + Send + Sync + 'static>;
-
-pub(crate) struct RequestHandler<Req, Resp> {
+pub struct GenericRequestHandler<Request: GenericRequest> {
     request_receiver: mpsc::Receiver<IncomingRequest>,
-    request_handler: ExternalRequestHandler<Req, Resp>,
-    log_target: &'static str,
-    protocol_name: &'static str,
+    #[allow(clippy::type_complexity)]
```

Comment:
Please, introduce a type alias here.

Reply:
It was just for one line and would look equally complex, that is the reason why I decoded to not create a type alias for it